compiler-plugin/src/main/java/io/ballerina/lib/np/compilerplugin/ExpressionValidator.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.lib.np.compilerplugin;
+
+import io.ballerina.compiler.api.SemanticModel;
+import io.ballerina.compiler.api.Types;
+import io.ballerina.compiler.api.symbols.Symbol;
+import io.ballerina.compiler.api.symbols.TypeDefinitionSymbol;
+import io.ballerina.compiler.api.symbols.TypeSymbol;
+import io.ballerina.compiler.syntax.tree.FunctionArgumentNode;
+import io.ballerina.compiler.syntax.tree.FunctionCallExpressionNode;
+import io.ballerina.compiler.syntax.tree.NaturalExpressionNode;
+import io.ballerina.compiler.syntax.tree.Node;
+import io.ballerina.compiler.syntax.tree.ParenthesizedArgList;
+import io.ballerina.compiler.syntax.tree.SeparatedNodeList;
+import io.ballerina.projects.Document;
+import io.ballerina.projects.Module;
+import io.ballerina.projects.ModuleId;
+import io.ballerina.projects.Package;
+import io.ballerina.projects.plugins.AnalysisTask;
+import io.ballerina.projects.plugins.SyntaxNodeAnalysisContext;
+import io.ballerina.tools.diagnostics.Location;
+
+import java.util.Optional;
+
+import static io.ballerina.lib.np.compilerplugin.Commons.MODULE_NAME;
+import static io.ballerina.lib.np.compilerplugin.Commons.ORG_NAME;
+import static io.ballerina.lib.np.compilerplugin.Commons.VERSION;
+import static io.ballerina.lib.np.compilerplugin.Commons.isNotNPCallCall;
+import static io.ballerina.lib.np.compilerplugin.DiagnosticLog.DiagnosticCode.EXPECTED_A_SUBTYPE_OF_NP_MODEL;
+import static io.ballerina.lib.np.compilerplugin.DiagnosticLog.DiagnosticCode.NON_JSON_EXPECTED_TYPE_NOT_YET_SUPPORTED;
+import static io.ballerina.lib.np.compilerplugin.DiagnosticLog.DiagnosticCode.UNEXPECTED_ARGUMENTS;
+import static io.ballerina.lib.np.compilerplugin.DiagnosticLog.reportError;
+
+/**
+ * Natural programming function signature validator.
+ *
+ * @since 0.3.0
```

Comment:
```suggestion
 * @since 0.3.0
```

Is this version correct

Reply:
This is from moved code (previously NPFunctionValidator), we can update in one go if necessary, before the release.